packages/@aws-cdk/aws-stepfunctions-tasks/test/emrcontainers/start-job-run.test.ts

```diff
@@ -0,0 +1,1173 @@
+  });
+
+  test('Tags', () => {
+
+    // WHEN
+    const task = new EmrContainersStartJobRun(stack, 'EMR Containers Start Job Run', {
+      virtualCluster: VirtualClusterInput.fromTaskInput(sfn.TaskInput.fromText(clusterId)),
+      releaseLabel: ReleaseLabel.EMR_6_2_0,
+      jobDriver: {
+        sparkSubmitJobDriver: {
+          entryPoint: sfn.TaskInput.fromText('local:///usr/lib/spark/examples/src/main/python/pi.py'),
+          sparkSubmitParameters: '--conf spark.executor.instances=2',
+        },
+      },
+      tags: {
+        key: 'value',
+      },
+    });
+
+    // THEN
+    expect(stack.resolve(task.toStateJson())).toMatchObject({
+      Parameters: {
+        Tags: {
+          key: 'value',
+        },
+      },
+    });
+  });
+
+
+  test('Application Configuration', () => {
+
+    // WHEN
+    const task = new EmrContainersStartJobRun(stack, 'EMR Containers Start Job Run', {
+      virtualCluster: VirtualClusterInput.fromTaskInput(sfn.TaskInput.fromText(clusterId)),
+      releaseLabel: ReleaseLabel.EMR_6_2_0,
+      jobDriver: {
+        sparkSubmitJobDriver: {
+          entryPoint: sfn.TaskInput.fromText('local:///usr/lib/spark/examples/src/main/python/pi.py'),
+          sparkSubmitParameters: '--conf spark.executor.instances=2',
+        },
+      },
+      applicationConfig: [{
+        classification: Classification.SPARK_DEFAULTS,
+        properties: {
+          'spark.executor.instances': '1',
+          'spark.executor.memory': '512M',
+        },
+      }],
+    });
+
+    // THEN
+    expect(stack.resolve(task.toStateJson())).toMatchObject({
+      Parameters: {
+        ConfigurationOverrides: {
+          ApplicationConfiguration: [{
+            Classification: Classification.SPARK_DEFAULTS.classificationStatement,
+            Properties: {
+              'spark.executor.instances': '1',
+              'spark.executor.memory': '512M',
+            },
+          }],
+        },
+      },
+    });
+  });
+
+  test('Job Execution Role', () => {
+
+    // WHEN
+    const task = new EmrContainersStartJobRun(stack, 'EMR Containers Start Job Run', {
+      virtualCluster: VirtualClusterInput.fromTaskInput(sfn.TaskInput.fromText(clusterId)),
+      releaseLabel: ReleaseLabel.EMR_6_2_0,
+      executionRole: iam.Role.fromRoleArn(stack, 'Job Execution Role', 'arn:aws:iam::xxxxxxxxxxxx:role/JobExecutionRole'),
+      jobDriver: {
+        sparkSubmitJobDriver: {
+          entryPoint: sfn.TaskInput.fromText('local:///usr/lib/spark/examples/src/main/python/pi.py'),
+          sparkSubmitParameters: '--conf spark.executor.instances=2',
+        },
+      },
+    });
+
+    // THEN
+    expect(stack.resolve(task.toStateJson())).toMatchObject({
+      Parameters: {
+        ExecutionRoleArn: 'arn:aws:iam::xxxxxxxxxxxx:role/JobExecutionRole',
+      },
+    });
+
+    // THEN
+    expect(stack.resolve(task.toStateJson())).toEqual(expect.not.objectContaining({
+      Type: 'AWS::CloudFormation::CustomResource',
+    }));
```

Comment:
not 100% sure on syntax
```suggestion
    expect(stack).not.toHaveResource('AWS::CloudFormation::CustomResource');
```